### DISCUSSION TITLE
怎么理解IOP，以及如何用它构建proof system？

### DISCUSSION THREAD

**Discussion by Howard-Hu** [zkp-co-learning/zkp-co-learn#42] [Q&A（提问题在隔壁~）]
什么是IOP？
如何用它来构建proof system?
它和commitment方案的关系是什么？

**Comment by Demian101** [comment]
信息理论交互式 Oracle Proof（IOP）和交互式证明（Interactive proof）不完全相同 :

 - 交互式证明（Interactive proof）是指一个交互式的证明协议，其中一个证明者（称为证明者）试图向另一个实体（称为验证者）证明某个命题的真实性，通常是一个数学定理或某个计算问题的解。证明者通过在交互过程中向验证者发送一系列消息（例如数字、文本或计算结果），并基于验证者返回的响应来构建他们的证明。交互式证明在计算复杂性理论和密码学中都有广泛的应用。
 - 信息理论交互式 Oracle 证明（IOP）则是一种特殊类型的交互式证明，其中证明者和验证者都与一个称为 Interactive Oracle 的第三方实体交互。Interactive Oracle 是一个黑盒程序，可以回答一些特定的查询，例如回答某个输入字符串的某个位置上的字符是什么 (安比实验室 intro 第三课被绑架的天使 😇 ) 。在 IOP 中，证明者和验证者都可以向 Interactive Oracle发送查询，并基于 Oracle 的响应来构建他们的证明

因此，IOP是交互式证明的一种特殊形式，其中引入了一个额外的理想化实体（交互式 Oracle）来协助证明过程。

**Reply by Demian101** [reply]
这个只回答了你部分问题，我也蹲其余 2 个的答案
